### ISSUE TITLE
False-positive invalid-name (UPPER_CASE naming style) when assigning a global function

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#3454]
### Steps to reproduce
```
pylint bug.py
```

with `bug.py`:
```
"""Demonstrate false-positive invalid-name when assigning a global function."""


def my_function():
    """Do nothing."""


my_other_function = my_function
```

### Current behavior
`bug.py:8:0: C0103: Constant name "my_other_function" doesn't conform to UPPER_CASE naming style (invalid-name)`

### Expected behavior
No error or warning. `my_other_function` is still a function, not a constant.

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.2 (default, Feb 26 2020, 09:52:54) 
[GCC 7.5.0]
```

**Comment by PCManticore** [comment]
This is no longer reproduceable with the unreleased version, both installed from the `master` branch, namely:
```
pylint 2.5.0-dev1
astroid 2.4.0
```

Please give it a go and let us know if you can still reproduce it.
